plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/gramine/GenerateGramineSGXManifest.kt

```diff
@@ -0,0 +1,96 @@
+    companion object {
+        const val GRAMINE_SGX_SIGN_EXECUTABLE = "gramine-sgx-sign"
+        const val GRAMINE_GET_TOKEN_EXECUTABLE = "gramine-sgx-get-token"
+    }
+
+    @get:InputFile
+    val privateKey: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val directManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    val enclaveJar: RegularFileProperty = objects.fileProperty()
+
+    @get:InputFile
+    @Optional
+    val pythonSourcePath: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxManifest: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxToken: RegularFileProperty = objects.fileProperty()
+
+    @get:OutputFile
+    val sgxSig: RegularFileProperty = objects.fileProperty()
+
+    override fun action() {
+        val manifestPath = directManifest.get().asFile.absolutePath
+        val privateKeyPath = privateKey.get().asFile.absolutePath
+        val outputSGXManifestPath = sgxManifest.asFile.get().parentFile.absolutePath
+        val enclaveDestinationJarName = "${outputSGXManifestPath}/$GRAMINE_ENCLAVE_JAR"
+        val enclaveDestinationJarFile = File(enclaveDestinationJarName)
+        checkNotNull(
+            enclaveJar.get().asFile.copyTo(
+                enclaveDestinationJarFile,
+                overwrite = true
+            )
+        ) { "Enclave jar file not copied correctly while builgetEnclaveThreadCountFromManifestding the Gramine SGX manifest" }
+        val enclaveDestinationPythonFileName = "${outputSGXManifestPath}/$PYTHON_FILE"
+        val enclaveDestinationPythonFile = File(enclaveDestinationPythonFileName)
+
+        if (pythonSourcePath.isPresent) {
+            checkNotNull(
+                pythonSourcePath.get().asFile.copyTo(
+                    enclaveDestinationPythonFile,
+                    overwrite = true
+                )
+            ) { "Python script not copied correctly while building the Gramine SGX manifest" }
+        }
+
+        if (signDirectManifest(manifestPath, privateKeyPath).exitValue != 0) {
+            throw GradleException("Could not sign the manifest")
+        }
+
+        if (sgxGetToken().exitValue != 0) {
+            throw GradleException("Could not get the token for the SGX manifest")
+        }
+    }
+
+    private fun signDirectManifest(directManifest: String, privateKey: String): ExecResult {
+        val command = listOf(
+            GRAMINE_SGX_SIGN_EXECUTABLE,
+            "--manifest=$directManifest",
+            "--key=$privateKey",
+            "--output=${sgxManifest.asFile.get().absolutePath}"
+        )
+        val enclaveJarBuildDir = sgxManifest.asFile.get().parentFile.absolutePath
+        return commandLine(command, commandLineConfig = CommandLineConfig(workingDir = enclaveJarBuildDir))
+    }
+
+    private fun sgxGetToken(): ExecResult {
```

Comment:
What is this SGX token? There doesn't seem to be any information in their docs.

Reply:
https://gramine.readthedocs.io/en/stable/devel/building.html#prepare-a-signing-key
My understanding is that this is related to the SGX EINITTOKEN. Is this something that was needed for EPID? It looks like we can get rid of it actually.
All the examples I have met are using it though.

Reply:
It does look like it's related to EPID. Remove it if that's the case.

Reply:
Removed

Reply:
Readded! Getting a failure otherwise. It is not harming, so I would keep it.

Reply:
I wonder if setting `remote_attestation` to DCAP will remove the need for this.